src/Tracking/TrackJobScheduler.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace Give\Tracking;
+
+use Give\Tracking\Repositories\TrackEvents;
+
+/**
+ * Class TrackJobScheduler
+ * @package Give\Tracking
+ * @since 2.10.0
+ */
+class TrackJobScheduler {
+	/**
+	 * @var TrackRegisterer
+	 */
+	private $track;
+
+	/**
+	 * @var TrackEvents
+	 */
+	private $trackEvents;
+
+	/**
+	 * TrackJobScheduler constructor.
+	 *
+	 * @param  TrackRegisterer  $track
+	 * @param  TrackEvents  $trackEvents
+	 */
+	public function __construct( TrackRegisterer $track, TrackEvents $trackEvents ) {
+		$this->track       = $track;
+		$this->trackEvents = $trackEvents;
+	}
+
+	/**
+	 * Schedule cron job to send request to telemetry server.
+	 *
+	 * @since 2.10.0
+	 */
+	public function schedule() {
+		if ( ! $this->track->hasNewTracks() ) {
+			return;
+		}
+
+		$hookName = $this->getCronJobHookName();
+		$this->trackEvents->saveTrackList();
+		if ( ! wp_next_scheduled( $hookName ) ) {
+			wp_schedule_single_event( strtotime( 'tomorrow midnight', current_time( 'timestamp' ) ), $hookName );
```

Comment:
If we schedule the same time for every site then we're going to get a lot of requests from every timezone at the same time — that kind of concurrency will be hard the telemetry server. So I suggest that instead of being at midnight, simply make it `+24 hours` from the current time. It's also not a heavy request to send, so I'm not worried about it bogging anything down during peak hours.

Reply:
fixed: https://github.com/impress-org/givewp/compare/91a13db1b474...1b03f7ff33cc